typegate/src/utils.ts

```diff
@@ -214,3 +207,74 @@ export function collectFieldNames(tg: TypeGraph, typeIdx: number) {
   }
   return { title: typ?.title, fields: [] };
 }
+
+/**
+ * base64 decode and untar at cwd/{dir}
+ */
+export async function uncompress(dir: string, tarb64: string) {
+  const baseDir = path.join(Deno.cwd(), dir);
+  const buffer = base64.decode(tarb64);
+  const streamReader = new Blob([buffer])
+    .stream()
+    .pipeThrough(new DecompressionStream("gzip"))
+    .getReader();
+  const denoReader = streams.readerFromStreamReader(streamReader);
+
+  const untar = new Untar(denoReader);
+  const entries = [];
+  for await (const entry of untar) {
+    entries.push(entry.fileName);
+    if (entry.fileName == ".") {
+      continue;
+    }
+    let file: Deno.FsFile | undefined;
+    try {
+      if (entry.type === "directory") {
+        const resDirPath = path.join(baseDir, entry.fileName);
+        await ensureDir(resDirPath);
+        continue;
+      }
+      const resFilePath = path.join(baseDir, entry.fileName);
+      await ensureFile(resFilePath);
+
+      file = await Deno.open(resFilePath, { write: true });
+      await streams.copy(entry, file);
+    } catch (e) {
+      throw e;
+    } finally {
+      file?.close();
+    }
+  }
+  return baseDir;
+}
+
+export async function structureRepr(str: string): Promise<FolderRepr> {
```

Comment:
The function name lacks of context, though the function is placed in the common `utils.ts` file...